docs/2.9.0/docs/canton/usermanual/upgrading.rst

```diff
@@ -514,28 +581,34 @@ You can do that by either controlling your applications, or by
     .. assert:: { participant.domains.connect_local(olddomain); true }
     .. success:: participant.resources.set_resource_limits(ResourceLimits(Some(0), Some(0)))
 
-This will reject all commands and finish processing the pending commands. Once you are sure that
-your participant node is idle, disconnect the participant node from the old sync domain
-connection:
+This rejects all commands and finishes processing the pending commands. Once you are sure that
+your participant node is idle, disconnect the participant node from the old sync domain:
 
 .. snippet:: migrating_protocol
     .. success:: participant.domains.disconnect("olddomain")
 
-Test that the sync domain is disconnected by checking the list of active connections:
+Test that the participant is disconnected from the sync domain by checking the list of active connections:
 
 .. snippet:: migrating_protocol
     .. success:: participant.domains.list_connected()
     .. assert:: participant.domains.list_connected().isEmpty
 
-This is a good time to perform a backup of the database before proceeding:
+This is a good time to perform a backup of the database before proceeding.
 
 .. code:: sql
 
     CREATE DATABASE newdb WITH TEMPLATE originaldb OWNER dbuser;
 
+.. warning::
```

Comment:
The fact that the warning shows before the invocation of the command but refers to it makes the reading a bit difficult. I'd suggest to rephrase slightly:
```
.. warning::

    Note that the hard domain migration modifies the participant's data storage.
    Without a database backup for your participant a potential recovery becomes significantly
    more difficult.
```
or even
```
.. warning::

    Note that the following steps modify participant's data storage.
    Without a database backup for your participant a potential recovery becomes significantly
    more difficult.
```